### PR TITLE
add an 'init' command

### LINKED ISSUE #76: command to initialize tool/grinder.dart
It would be cool to do:

```shell
stagehand ...
den install grinder
pub run grinder:init
```

To initialize a grinder project.  `init` would probably not need to be in `executables`, it would initialize a default `tool/grind(er).dart` script.

### PR DESCRIPTION
**Pull request by devoncarew**
Fix #76; @seaneagan

### CODE REVIEW

File: bin/init.dart [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ */
+void main(List args) {
+  File file = new File('tool${Platform.pathSeparator}grind.dart');
+
+  if (file.existsSync()) {
+    print('Error: ${file.path} already exists.');
+    exit(1);
+  }
+
+  file.writeAsStringSync(_grindSampleSource);
```
Comment (seaneagan) [inline bin/init.dart]: I think `writeAsStringSync` will create parent directories (`tool` here) if they don't exist, but the docs don't say.

File: bin/init.dart [hunk 2]
```diff
@@ -0,0 +1,36 @@
+// Copyright 2015 Google. All rights reserved. Use of this source code is
+// governed by a BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+/**
+ * A command to create a simple, starting `tool/grind.dart` build script.
+ */
+void main(List args) {
+  File file = new File('tool${Platform.pathSeparator}grind.dart');
```
Comment (seaneagan) [inline bin/init.dart]: I think this should either validate that we are at the root of a pub package (`pubspec.yaml` exists) or try to climb the file system until a `pubspec.yaml` is found and produce a path relative to that.

File: bin/init.dart [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  if (file.existsSync()) {
+    print('Error: ${file.path} already exists.');
+    exit(1);
+  }
+
+  file.writeAsStringSync(_grindSampleSource);
+  print('Wrote ${file.path}!');
+}
+
+final String _grindSampleSource = '''
```
Comment (seaneagan) [inline bin/init.dart]: Probably should add a library name e.g. `library $packageName.tool.grind;`.

Comment (seaneagan) [inline bin/init.dart]: If so, then we should update the examples too.

File: bin/init.dart (hunk 2 above)
Comment (devoncarew) [inline bin/init.dart]: I think we should validate and fail otherwise; I'll add that.

File: bin/init.dart [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import 'package:grinder/grinder.dart';
+
+main(args) => grind(args);
+
+@Task()
+init(GrinderContext context) => defaultInit(context);
```
Comment (seaneagan) [inline bin/init.dart]: There is support for using properties:

```dart
@Task()
const init = defaultInit;
```

What do you think?

File: bin/init.dart (hunk 1 above)
Comment (devoncarew) [inline bin/init.dart]: Ah, I'll explicitly create the `tool` dir first.

File: readme.md [hunk 5]
```diff
@@ -15,8 +15,11 @@ source.
 
 ## Getting Started
 
-Your grinder build file should reside at `tool/grind.dart`, and may look
-something like:
+Your grinder build file should reside at `tool/grind.dart`. You can use grinder
+to create a simple, starting build script. Run `pub run grinder:init`; this will
```
Comment (seaneagan) [inline readme.md]: nit: might be worth putting `pub run grinder:init` on its own line, just so it stands out.

File: bin/init.dart (hunk 4 above)
Comment (seaneagan) [inline bin/init.dart]: scratch that, it doesn't help the uninitiated user figure out how to define arbitrary tasks.

**Comment by seaneagan** [comment]
lgtm!

File: readme.md (hunk 5 above)
Comment (devoncarew) [inline readme.md]: done

**Comment by devoncarew** [comment]
- updated readme
- added a library name to the `init.dart` script
- check for running from the project root

File: bin/init.dart (hunk 4 above)
Comment (seaneagan) [inline bin/init.dart]: Once #78 is fixed, it may make more sense though.

Comment (devoncarew) [inline bin/init.dart]: I like simple and explicit for new users, at the expense of being slightly more verbose.

**Pull request merged**